tempodb/backend/azure/azure_helpers.go

```diff
@@ -181,11 +201,37 @@ func getOAuthToken(cfg *Config) (*blob.TokenCredential, error) {
 	return &tc, nil
 }
 
-func getServicePrincipalToken(cfg *Config) (*adal.ServicePrincipalToken, error) {
+func getServicePrincipalToken(cfg *Config, authFunctions authFunctions) (*adal.ServicePrincipalToken, error) {
```

Comment:
it seems like for `getServicePrincipalToken()` we always pass `defaultAuthFunctions`. Can we remove the `authFunctions` parameter here? and only take it at `servicePrincipalTokenFromFederatedToken` level?

Reply:
Done in https://github.com/grafana/tempo/pull/2195/commits/83b5d32b2d091cd8646ef047d99c26d802e6d8ce